aws_list_all/generate_html.py

```diff
@@ -0,0 +1,225 @@
+    print('@keyframes fadeIn {from {opacity: 0;} to {opacity:1 ;}}\n')
+
+    print('#searchInput {\n')
+    print('  background-position: 10px 10px;\n')
+    print('  background-repeat: no-repeat;\n')
+    print('  width: 300px;\n')
+    print('  font-size: 16px;\n')
+    print('  padding: 12px 20px 12px 40px;\n')
+    print('  border: 1px solid #ddd;\n')
+    print('  margin-bottom: 12px;\n')
+    print('}\n')
+    print('</style>\n')
+    print('</head>\n')
+
+
+def generate_header():
+    print('<div class="headerbar">')
+    print('  <a href="https://github.com/kntrain/aws_list_all"><b>aws_list_all</b></a>')
+    print('  <div class="searchbar">')
+    print('    <input type="text" id="searchInput" onkeyup="search()" placeholder="FIlter for ...">')
+    print('  </div>')
+    print('</div>')
+    print('<br>')
+
+
+def generate_table(results_by_region, services_in_grid):
+    print('<div class="main">')
+    print('<table class="aws-table"; id="mainTable"; table-layout:fixed;>')
+    print('    <tr>\n')
+    print('        <th>Service</th>\n')
+    for region_column in sorted(results_by_region):
+        print('<th >' + str(region_column) + '</th>\n')
+    print('    </tr>\n')
+    rest_by_type = defaultdict(list)
+    for service_type in sorted(services_in_grid):
+        print('    <tr>\n')
+        print('        <td class="service">' + service_type + '</td>\n')
+        for result_region in sorted(results_by_region):
+            print('        <td width="450">\n')
+            for result_type in ('---', '+++', '>:|', '!!!'):
+                empty_type = True
+                result_type_list = list(filter(lambda x: x.input.service == service_type, sorted(results_by_region[result_region][result_type])))
+                result_type_count = ' [' + str(len(result_type_list)) + ']'
+                for result in result_type_list:
+                    if empty_type:
+                        print('        <button type="button" class="' + status_switch(result_type + 'col') + '">'
+                            + status_switch(result_type + 'box') + result_type_count + '</button>\n')
+                        print('        <div class="content">\n')
+                        empty_type = False
+                    diff_color = ''
+                    if result.diff:
+                        diff_color = 'style="background-color:' + status_switch(result.diff) + ';"'
+                    print('<div class="' + status_switch(result_type) + '" ' + diff_color + '>')
+                    print(str(result.input.operation))
+                    if result_type == '+++':
+                        print('<p style="font-size:14px;margin:0">')
+                        print(str(result.id_list))
+                        print('</p>')
+                    print('</div>')
+                if not(empty_type):
+                    print('        </div>')
+            print('        </td>')
+        print('    </tr>\n')
+    print('</table>')
+    print('</div>')
```

Comment:
I am tempted to suggest a more generic solution here, but this is actually quite readable.